tests/publiccloud/run_ltp.pm

```diff
@@ -121,15 +125,67 @@ sub cleanup {
     type_string('', terminate_with => 'ETX');
 
     upload_logs('ltp_log.raw', failok => 1);
-    parse_extra_log(LTP => "$root_dir/ltp_log.json") if (script_run("test -f $root_dir/ltp_log.json") == 0);
-
+    if (script_run("test -f $root_dir/ltp_log.json") == 0) {
+        my $known_issues = download_whitelist();
+        process_ltp_known_issues("$root_dir/ltp_log.json") if $known_issues;
+        parse_extra_log(LTP => "$root_dir/ltp_log.json");
+    }
     if ($self->{my_instance} && script_run("test -f $root_dir/log_instance.sh") == 0) {
         assert_script_run($root_dir . '/log_instance.sh stop ' . $self->instance_log_args());
         assert_script_run("(cd /tmp/log_instance && tar -zcf $root_dir/instance_log.tar.gz *)");
         upload_logs("$root_dir/instance_log.tar.gz", failok => 1);
     }
 }
 
+sub process_ltp_known_issues {
+    my ($self, $ltp_log) = @_;
+    my $decoded_ltp_log = Mojo::JSON::from_json($ltp_log);
+    my $env             = gen_ltp_env();
+    my $suite           = get_required_var('COMMAND_FILE');
+
+    foreach my $test (@{$decoded_ltp_log->{results}}) {
+        my @issues;
+        if (($test->{'status'} eq 'fail' || $test->{'status'} eq 'brok' || $test->{'status'} eq 'warn') && ref($known_issues->{$suite}) eq 'ARRAY') {
+            my $entry = find_whitelist_entry($env, $suite, $test->{test_fqn});
+            bmwqemu::diag(sprintf("Failure in LTP:%s:%s is known, overriding to softfail", $suite, $test->{test_fqn}));
+            $test->{status} = 'softfail';
+        }
+    }
+    my $final_ltp_log = Mojo::JSON::to_json($decoded_ltp_log);
+    Mojo::File::path($ltp_log)->spurt($final_ltp_log);
+}
+
+sub gen_ltp_env {
+    my $environment = {
+        product     => get_required_var('DISTRI') . ':' . get_required_var('VERSION'),
+        revision    => get_required_var('BUILD'),
+        flavor      => get_required_var('FLAVOR'),
+        arch        => get_var('PUBLIC_CLOUD_ARCH', get_required_var("ARCH")),
+        backend     => get_required_var('BACKEND'),
+        kernel      => '',
+        libc        => '',
+        gcc         => '',
+        harness     => 'SUSE OpenQA',
+        ltp_version => '',
+        #TODO currently we getting only pass or fail due to use of '--openqa-filter' need to properly solve retval issue
+        # from runltp-ng side to get smarter logic on this side
+        retval => '1'
```

Comment:
currently due to this simplification https://github.com/metan-ucw/runltp-ng/blob/master/results.pm#L458 we are not getting anything except "pass"/"fail" from runltp-ng . and before we will fix this such hard-coded value is our only option ( because here https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12734/files#diff-49293c308cb7b8e8f6bbfc7f14a6fa90507cd690aa9d5ab41c1f60f96e4cb5ecR148 we filtering "pass" tests anyway )

Reply:
It'd be better to do this in `process_ltp_known_issues()` instead:

    %resultcodes = {
    	pass => 0,
    	fail => 1,
    	brok => 2,
    	warn => 4,
    	skip => 32
    };
    
    ...
    
    $env->{retval} = $resultcodes{$test->{'status'}} // 'undefined';